lib/cadet/incentives/achievements.ex

```diff
@@ -21,7 +21,43 @@ defmodule Cadet.Incentives.Achievements do
     |> Repo.all()
   end
 
-  @spec upsert(map()) :: {:ok, Achievement.t()} | {:error, {:bad_request, String.t()}}
+  @doc """
+  Returns all achievements with goals and progress.
+  # To Do
+  Run elixir tests for edge cases etc.
+
+  This returns Achievement structs with prerequisites, goal association and progress maps pre-loaded.
+  """
+  def get_total_xp(course_id) when is_ecto_id(course_id) do
```

Comment:
This function is a bit strange. Is it supposed to return the total possible XP of all achievements in a course?